src/code_8069D4C.c

```diff
@@ -1,8 +1,394 @@
+    {
+        tile = GetTile((target->pos).x + (gAdjacentTileOffsets)[info->action.direction].x,(target->pos).y + (gAdjacentTileOffsets)[info->action.direction].y);
+        return tile->monster;
+    }
+    return NULL;
+}
+
+Entity * sub_80696A8(Entity *target)
+{
+    const Tile *tile;
+    Entity *entity;
+
+    EntityInfo *info;
+
+    info = GetEntInfo(target);
+
+    if(CanAttackInDirection(target, info->action.direction))
+    {
+        tile = GetTile((target->pos).x + (gAdjacentTileOffsets)[info->action.direction].x,(target->pos).y + (gAdjacentTileOffsets)[info->action.direction].y);
+        if (((entity = tile->monster, entity != NULL)) && (entity->type == ENTITY_MONSTER)) {
+            return entity;
+        }
+    }
+    return NULL;
+}
+
+
+Entity * sub_80696FC(Entity *target)
+{
+    const Tile *tile;
+    Entity *entity;
+
+    DungeonPos pos;
+    EntityInfo *info;
+
+    info = GetEntInfo(target);
+
+    pos.x = (target->pos).x + (gAdjacentTileOffsets)[info->action.direction].x;
+    pos.y =  (target->pos).y + (gAdjacentTileOffsets)[info->action.direction].y;
+    tile = GetTile(pos.x,pos.y);
+    if (((!sub_80705F0(target,&pos)) && (entity = tile->monster, entity != NULL)) && (entity->type == ENTITY_MONSTER)) {
+        return entity;
+    }
+    return NULL;
+}
+
+Entity * sub_806977C(Entity *target)
+{
+    const Tile *tile;
+    Entity *entity;
+
+    DungeonPos pos;
+    EntityInfo *info;
+
+    info = GetEntInfo(target);
+
+    pos.x = (target->pos).x + (gAdjacentTileOffsets)[info->action.direction].x;
+    pos.y =  (target->pos).y + ((gAdjacentTileOffsets)[info->action.direction].y);
+    GetTile(pos.x,(int)pos.y);
+    if (!sub_80705F0(target,&pos)) {
+        pos.x = (gAdjacentTileOffsets)[info->action.direction].x * 2 + (target->pos).x;
+        pos.y = (gAdjacentTileOffsets)[info->action.direction].y * 2 + (target->pos).y;
+        tile = GetTile(pos.x,pos.y);
+        if (((!sub_80705F0(target,&pos)) && (entity = tile->monster, entity != NULL)) && (entity->type == ENTITY_MONSTER)) {
+            return entity;
+        }
+    }
+    return NULL;
+}
+
+void sub_8069844(struct unkStruct_808FF20 *param_1, Entity *target) {
+    int index;
+    u8 *atkPtr;
+    u8 *spAtkPtr;
+    u8 *defPtr;
+    s32 uVar15;
+    u8 *spDefBoost;
+
+    EntityInfo *info = GetEntInfo(target);
+
+    SetMessageArgument_2(param_1->nameBuffer, info, 0);
+    param_1->species = info->id;
+    param_1->HP1 = info->HP;
+    param_1->HP2 = info->maxHPStat;
+    param_1->level = info->level;
+    param_1->exp = info->exp;
+    for(index = 0; index < 2; index++)
+    {
+        (param_1->offense).att[index] = info->atk[index];
+        (param_1->offense).def[index] = info->def[index];
+        param_1->types[index] = info->types[index];
+        param_1->abilities[index] = info->abilities[index];
+    }
+    param_1->isTeamLeader = info->isTeamLeader;
+    atkPtr = &param_1->atkBoost;
+    *atkPtr = '\0';
```

Comment:
Stat boosts are numerical values, so 0 would make more sense than the null terminator.